### PR TITLE
Replace costura with dotnet single file publish

### PR DESCRIPTION
**Pull request by EbenZhang**
#38 should just make the choco package work.

This is an additional pr to replace the costura with dotnet single file publishing.
I'm not very sure about the appveyor and choco part, so maybe you want to verify it a bit before publishing.

I have tested the file on windows 7 x86 without dotnet core runtime. 

this pr also including the changes in #38, which will disappear once #38 merged.

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -10,27 +10,19 @@ platform:
...
 
 build_script:
   - cmd: nuget restore src/PullRequestReleaseNotes.sln
-  - ps: gitversion /l console /output buildserver /updateAssemblyInfo 
-  - cmd: msbuild src/PullRequestReleaseNotes.sln "/p:Configuration=%CONFIGURATION%;Platform=%PLATFORM%" 
-  - cmd: xcopy /S /Y src\PullRequestReleaseNotes\bin\Debug\net472 build\NuGetExeBuild\net\tools\
+  - ps: gitversion /l console /output buildserver /updateAssemblyInfo
+  - ps: dotnet pack .\src\PullRequestReleaseNotes\PullRequestReleaseNotes.csproj "/p:Configuration=$($env:CONFIGURATION);Platform=$($env:PLATFORM);Version=$($env:GitVersion_NuGetVersion)"
+  - ps: dotnet publish .\src\PullRequestReleaseNotes\PullRequestReleaseNotes.csproj --framework netcoreapp3.1 -p:PublishSingleFile=true --runtime win7-x86 "/p:Configuration=$($env:CONFIGURATION)"
   - cmd: xcopy /S /Y src\PullRequestReleaseNotes\NuGetAssets build\NuGetExeBuild\net\
```
Comment (jasminsehic) [inline appveyor.yml:20]: Yeah I am going to need to test this out on Mac and Linux to make sure the native libgit libraries are referenced correctly.

**Review by jasminsehic — COMMENTED** [review]
@EbenZhang I have merged everything now, just going to do some testing with the builds tonight to check everything works OK on both Choco and .NET Core on Windows, Mac and Linux.

**Comment by jasminsehic** [comment]
@EbenZhang I think this also includes the .NET Core run-time as well?

**Comment by EbenZhang** [comment]
The choco package is bigger as it now includes the required dlls to run on win7-x86 (and onwards of course)

**Comment by jasminsehic** [comment]
@EbenZhang I've tested .NET package OK on Mac with .NET Core 3.1 so that looks good.
Need to test with Linux and WIndows still, plus Choco.

**Comment by jasminsehic** [comment]
@EbenZhang all tested now, both NET Core and Choco packages run OK on all platforms. I am publishing shortly.

**Pull request merged**